src/main/java/io/cdap/plugin/gcp/bigquery/sqlengine/SQLEngineAvroToStructuredTransformer.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.gcp.bigquery.sqlengine;
+
+import com.google.common.annotations.VisibleForTesting;
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.cdap.etl.api.engine.sql.SQLEngineException;
+import io.cdap.plugin.gcp.bigquery.source.BigQueryAvroToStructuredTransformer;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import javax.annotation.Nullable;
+
+/**
+ * Create StructuredRecords from GenericRecords when pulling records from BigQuery.
+ *
+ * The following transformations are supported:
+ * BigQuery INT64 -> Integer (with Overflow check)
+ * BigQuery FLOAT64 -> Float (with Overflow check)
+ *
+ * Otherwise, field conversion gets delegated to the {@link BigQueryAvroToStructuredTransformer}
+ */
+public class SQLEngineAvroToStructuredTransformer extends BigQueryAvroToStructuredTransformer
+  implements Serializable {
```

Comment:
why does this need to be serializable?

Reply:
This transformer gets sent from the driver into the workers to execute the read from GCS after an export. If not serializable, Spark will complain.